### PR TITLE
Framework agreements

### PR DESCRIPTION
**Pull request by robyoung**
[#105012790](https://www.pivotaltracker.com/story/show/105012790)

## Get rid of get_supplier helper function
Now that the admin frontend translates API errors into the appropriate Flask response this helper does not offer much. One change here is that the supplier_id is no longer validated. This is left up to the API.

## Add signed agreement download link
Add a link to the `admin-ccs-sourcing` supplier list that allows them to download signed agreement documents.

## listing page for signed framework agreements

This adds a page that lists signed framework agreements in reverse chronological order (most recent first). This allows the CCS sourcing team to easily find recently uploaded agreements that need action.

screen shot 2015-11-09 at 10 53 33

## Depends on
- [x] Sign off from Cath
- [ ] :sparkles: Functional tests :sparkles:

### CODE REVIEW

**Comment by TheDoubleK** [comment]
Is this going to change now that we're having a timestamp-ordered list page? Or does this all want to go in too? (If so then I'll review it)

File: app/main/views/agreements.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+from .. import main
+from ..auth import role_required
+from ... import data_api_client
+from . import get_template_data
+
+
+@main.route('/agreements/<framework_slug>', methods=['GET'])
+@login_required
+@role_required('admin-ccs-sourcing')
```
Comment (allait) [inline app/main/views/agreements.py]: This should probably be `role_required('admin', 'admin-ccs-sourcing')`, unless we have a reason to disallow this for admin.

Comment (robyoung) [inline app/main/views/agreements.py]: Cool, double checked with Cath and adding 'admin' is ok.

File: app/main/views/suppliers.py [hunk 2]
```diff
@@ -142,7 +156,10 @@ def update_supplier_name(supplier_id):
 def find_supplier_users():
     form = EmailAddressForm()
 
-    supplier = get_supplier()
+    if len(request.args.get('supplier_id', '')) == 0:
```
Comment (allait) [inline app/main/views/suppliers.py]: `if not request.args.get('supplier_id')`

**Comment by robyoung** [comment]
@TheDoubleK I've added that page now so it's ready for review.

**Comment by TheDoubleK** [comment]
CCS might like a more user-friendly date format...
screen shot 2015-11-09 at 16 06 57

File: app/main/views/suppliers.py [hunk 3]
```diff
@@ -62,6 +63,20 @@ def view_supplier_declaration(supplier_id, framework_slug):
         **get_template_data())
 
 
+@main.route('/suppliers/<supplier_id>/agreements/<framework_slug>/<document_name>', methods=['GET'])
+@login_required
+@role_required('admin', 'admin-ccs-sourcing')
+def download_agreement_file(supplier_id, framework_slug, document_name):
+    supplier = data_api_client.get_supplier(supplier_id)['suppliers']
+    agreements_bucket = s3.S3(current_app.config['DM_AGREEMENTS_BUCKET'])
+    path = get_agreement_document_path(framework_slug, supplier_id, supplier['name'], document_name)
```
Comment (TheDoubleK) [inline app/main/views/suppliers.py]: Does this need to get the supplier name from the declaration too? (Probably?)

File: app/templates/index.html [hunk 4]
```diff
@@ -100,5 +100,18 @@
 {% include "toolkit/browse-list.html" %}
 {% endwith %}
 
+{% if current_user.has_any_role('admin-ccs-sourcing') %}
```
Comment (TheDoubleK) [inline app/templates/index.html]: Is it worth having the dashboard link for regular admins too?  Probably not? I don't think Nicole should be getting involved with these at all.

**Comment by robyoung** [comment]
@TheDoubleK both of those points have been addressed now

**Comment by TheDoubleK** [comment]
Looks good to me :+1: - let's get it out and check it all works right on Preview (or Staging if no Preview data).

**Pull request merged**